mne/io/pick.py

```diff
@@ -784,24 +788,163 @@ def _pick_aux_channels(info, exclude='bads'):
+    return picks
+
+
+def _picks_str_to_idx(info, picks, exclude, with_ref_meg):
+    """Turn a list of str into ndarray of int."""
+    # special case for _picks_to_idx w/no info: shouldn't really happen
+    if isinstance(info, int):
+        raise ValueError('picks as str can only be used when measurement '
+                         'info is available')
+
+    #
+    # first: check our special cases
+    #
+
+    picks_generic = list()
+    if len(picks) == 1:
+        if picks[0] == 'all':
+            use_exclude = info['bads'] if exclude == 'bads' else exclude
+            picks_generic = pick_channels(info['ch_names'], info['ch_names'],
+                                          exclude=use_exclude)
+        elif picks[0] == 'data':
+            picks_generic = _pick_data_channels(info, exclude=exclude,
+                                                with_ref_meg=with_ref_meg)
+        elif picks[0] == 'data_or_ica':
+            picks_generic = _pick_data_or_ica(info, exclude=exclude)
+
+    #
+    # second: match all to channel names
+    #
+
+    bad_name = None
+    picks_name = list()
+    for pick in picks:
+        try:
+            picks_name.append(info['ch_names'].index(pick))
+        except ValueError:
+            bad_name = pick
+            break
+
+    #
+    # third: match all to types
+    #
+    bad_type = None
+    picks_type = list()
+    kwargs = dict(meg=False)
+    meg, fnirs = set(), set()
+    for pick in picks:
+        if pick in _PICK_TYPES_KEYS:
+            kwargs[pick] = True
+        elif pick in _MEG_CH_TYPES_SPLIT:
+            meg |= {pick}
+        elif pick in _FNIRS_CH_TYPES_SPLIT:
+            fnirs |= {pick}
+        else:
+            bad_type = pick
+            break
+    else:
+        # triage MEG and FNIRS, which are complicated due to non-bool entries
+        extra_picks = set()
+        if len(meg) > 0 and not kwargs.get('meg', False):
+            # easiest just to iterate
+            for use_meg in meg:
+                extra_picks |= set(pick_types(
+                    info, meg=use_meg, ref_meg=False, exclude=exclude))
+        if len(fnirs) > 0 and not kwargs.get('fnirs', False):
+            # if it has two entries, it's both, otherwise it's just one
+            kwargs['fnirs'] = True if len(fnirs) == 2 else list(fnirs)[0]
+        picks_type = pick_types(info, exclude=exclude, **kwargs)
+        if len(extra_picks) > 0:
+            picks_type = sorted(set(picks_type) | set(extra_picks))
+
+    #
+    # finally: ensure we have exactly one usable list
+    #
+    all_picks = (picks_generic, picks_name, picks_type)
+    any_found = [len(p) > 0 for p in all_picks]
+    if sum(any_found) == 0:
+        raise ValueError(
+            'picks (array-like of str) could not be interpreted as '
+            'channel names (no channel "%s"), channel types (no '
+            'type "%s"), or a generic type (just "all" or "data")'
+            % (bad_name, bad_type))
```

Comment:
minor nitpick: it seems that passing a list of channels that are not found would result in the channel name being shown as bad_name and also bad_type. This would be due to the fact that picks are not modified so channel name will qualify as `else:` in the type loop.

Reply:
I think we can live with this for now, it should give users enough to go on to fix the problem